bundles/org.openhab.core.io.websocket/src/main/java/org/openhab/core/io/websocket/audiopcm/PCMWebSocketConnection.java

```diff
@@ -0,0 +1,371 @@
+                    var messageType = WebSocketCommand.InputCommands.valueOf(cmd);
+                    switch (messageType) {
+                        case INITIALIZE -> {
+                            wsAdapter.onSpeakerConnected(this);
+                            JsonNode argsNode = rootMessageNode.get("args");
+                            var clientOptions = jsonMapper.treeToValue(argsNode, ConnectionOptions.class);
+                            var scheduledDisconnection = this.scheduledDisconnection;
+                            if (scheduledDisconnection != null) {
+                                scheduledDisconnection.cancel(true);
+                            }
+                            // update connection settings
+                            id = clientOptions.id;
+                            registerSpeakerComponents(id, clientOptions);
+                            sendClientCommand(new WebSocketCommand(WebSocketCommand.OutputCommands.INITIALIZED));
+                        }
+                        case ON_SPOT -> onRemoteSpot();
+                    }
+
+                } catch (IOException | IllegalStateException e) {
+                    logger.warn("Disconnecting client: {}", e.getMessage());
+                    disconnect();
+                }
+            }
+        } catch (JsonProcessingException e) {
+            logger.warn("Exception parsing JSON message: ", e);
+        }
+    }
+
+    @Override
+    public void onWebSocketError(@Nullable Throwable cause) {
+        logger.warn("WebSocket Error: ", cause);
+    }
+
+    @Override
+    public void onWebSocketClose(int statusCode, @Nullable String reason) {
+        this.session = null;
+        this.remote = null;
+        logger.debug("Session closed with code {}: {}", statusCode, reason);
+        wsAdapter.onClientDisconnected(this);
+        unregisterSpeakerComponents(id);
+    }
+
+    public void setSinkVolume(int value) {
+        if (initialized) {
+            sendClientCommand(
+                    new WebSocketCommand(WebSocketCommand.OutputCommands.SINK_VOLUME, Map.of("value", value)));
+        }
+    }
+
+    public void setSourceVolume(int value) {
+        if (initialized) {
+            sendClientCommand(
+                    new WebSocketCommand(WebSocketCommand.OutputCommands.SOURCE_VOLUME, Map.of("value", value)));
+        }
+    }
+
+    public @Nullable RemoteEndpoint getRemote() {
+        return this.remote;
+    }
+
+    public @Nullable Session getSession() {
+        return this.session;
+    }
+
+    public boolean isConnected() {
+        Session sess = this.session;
+        return sess != null && sess.isOpen();
+    }
+
+    private synchronized void registerSpeakerComponents(String id, ConnectionOptions clientOptions) throws IOException {
+        if (id.isBlank()) {
+            throw new IOException("Unable to register audio components");
+        }
+        String label = "UI (" + id + ")";
```

Comment:
I think it might be better to leave out the UI from the label here, rather use PCM WebSocket (id), as not only the UI might connect.